src/WolkBuilder.cpp

```diff
@@ -121,66 +155,92 @@ std::unique_ptr<Wolk> WolkBuilder::build() const
         }
     }
 
+    if ((m_configurationHandlerLambda == nullptr && m_configurationProviderLambda != nullptr) ||
```

Comment:
Will this work as expected if both are null?
Maybe check  !((handler || handlerLambda) && (provider || providerLambda)) or something similar to combine with check below and allow for instance for handler and providerLambda to be set instead of forcing for handler and provider to be of the same type

Reply:
If both are null it means interfaces are implemented and used as such instead of lambdas.
In actuation, and here, interface implementations have priority over lambda callbacks